core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/CoreTerminologyBroker.java

```diff
@@ -369,6 +375,36 @@ public boolean isEffectiveTimeSupported(final String terminologyId) {
 		throw new RuntimeException("Cannot find terminology with ID: '" + terminologyId + "'.");
 	}
 	
+	public SortedSet<String> getAffectedCodeSystemsForTeminology(final String terminologyId) {
+		Preconditions.checkNotNull(terminologyId, "terminologyId");
+		
+		if (!TERMINOLOGY_ID_TO_SHORTNAMES_CACHE.containsKey(terminologyId)) {
```

Comment:
🛑 Still not a properly initialized cache!

The if expression should not rely on the existence of the given `terminologyId` key.